app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -222,15 +235,60 @@ public void onGetUserAccount(UserAccount userAccount) {
 
     }
 
-    private void openNewSurvey(Activity activity) {
+    private void openNewSurvey(final Activity activity) {
+        Survey survey = createNewConnectSurvey();
+
+        IMainExecutor mainExecutor = new UIThreadExecutor();
```

Comment:
mainExecutor and AsyncExecutor are created three times in this class, can you optimize it?